project/Publishing.scala

```diff
@@ -116,6 +127,44 @@ object Publishing {
     )
   )
 
+  /**
+    * Install packages needed for debugging when shelled in to a running image.
+    *
+    * This includes:
+    * - The JDK, which includes tools like `jstack` not present in the JRE
+    * - The YourKit Java Profiler
+    * - Various Linux system & development utilities
+    *
+    * @return Instruction to run in the build
+    */
+  def installDebugFacilities: Instruction = {
+    import sbtdocker.Instructions
+
+    // It is optimal to use a single `Run` instruction to minimize the number of layers in the image.
+    // Do not be tempted to install the default JDK in the repositories, it's from Oracle.
+    //
+    // Documentation:
+    // - https://www.yourkit.com/docs/java-profiler/2024.3/help/docker_broker.jsp#setup
+    // - https://adoptium.net/installation/linux/#_deb_installation_on_debian_or_ubuntu
+    Instructions.Run(
+      """apt-get update -qq && \
+        |apt-get install -qq --no-install-recommends file gpg htop less nload unzip vim && \
```

Comment:
maybe also helpful to have `curl` and `jq` installed to test web connections?

Reply:
`curl` is already present, will add `jq` 👍